core/src/avm2/globals/XML.as

```diff
@@ -60,7 +60,10 @@ package {
         AS3 native function hasSimpleContent():Boolean;
         AS3 native function name():Object;
         AS3 native function setName(name:*):void;
-        AS3 native function namespace(prefix:String = null):*;
+        private native function namespace_internal_impl(hasPrefix:Boolean, prefix:String = null):*;
+        AS3 function namespace(...args):* {
```

Comment:
we should keep the FP-compatible signature and instead check the magic `arguments` variable's length.